src/node/inversify.config.ts

```diff
@@ -11,63 +11,16 @@
  */
 
 import 'reflect-metadata';
-import { Container, interfaces } from 'inversify';
-import { NodeDevWorkspaceTemplateApi } from './services/API/template-api';
+import { Container } from 'inversify';
 import {
-    ICheApi,
-    IDevWorkspaceApi,
-    IDevWorkspaceClientApi,
-    IDevWorkspaceTemplateApi,
-    IDevWorkspaceWatcher,
-    INodeConfig,
     INVERSIFY_TYPES
 } from '../types';
-import { NodeDevWorkspaceApi } from './services/API/workspace-api';
-import { NodeApi } from './services/API';
-import * as k8s from '@kubernetes/client-node';
-import { isInCluster } from './services/helpers';
-import { DevWorkspaceClient } from './client';
-import { NodeCheApi } from './services/API/che-api';
-import { NodeDevWorkspaceWatcher } from './services/API/workspace-watcher';
+import { DevWorkspaceClientFactory } from './client-factory';
 
 const container = new Container();
-container.bind(INVERSIFY_TYPES.IDevWorkspaceClient).to(DevWorkspaceClient).inSingletonScope();
-container.bind(INVERSIFY_TYPES.IDevWorkspaceNodeClientApi).to(NodeApi).inSingletonScope();
-container.bind(INVERSIFY_TYPES.IDevWorkspaceNodeApi).to(NodeDevWorkspaceApi).inSingletonScope();
-container.bind(INVERSIFY_TYPES.IDevWorkspaceNodeTemplateApi).to(NodeDevWorkspaceTemplateApi).inSingletonScope();
-container.bind(INVERSIFY_TYPES.IDevWorkspaceNodeCheApi).to(NodeCheApi).inSingletonScope();
-container.bind(INVERSIFY_TYPES.IDevWorkspaceWatcher).to(NodeDevWorkspaceWatcher).inSingletonScope();
+container.bind(INVERSIFY_TYPES.IDevWorkspaceClientFactory).to(DevWorkspaceClientFactory).inSingletonScope();
```

Comment:
My understanding of why we bound everything individually and then created the factory was so that implementations could swap out the underlying API. Now we just have the factory which makes me wonder why we need inversify anymore if doesn't really provide us any advantage

Reply:
you're right. Inversify does not make sense anymore. I'll remove it ATM

Reply:
Well, we still may keep in container DevWorkspaceClient instance that will be based on autodetected configuration (in-cluster, or kubeconfig if we run it locally) but clients must understand the scope where they should use it.

Reply:
but at the same time clients may have container with bound kubeconfig, which will be more explicit and less error-prone. I'll think about it more

Reply:
Removed inversify and factory at all.
clients needs to control their needed kubeconfigs and resolve them in their own way.